onnxruntime.sh

```diff
@@ -19,24 +19,71 @@ prepend_path:
 
 mkdir -p $INSTALLROOT
 
-cmake "$SOURCEDIR/cmake"                                                              \
-      -DCMAKE_INSTALL_PREFIX=$INSTALLROOT                                             \
-      -DCMAKE_BUILD_TYPE=Release                                                      \
-      -DCMAKE_INSTALL_LIBDIR=lib                                                      \
-      -DPYTHON_EXECUTABLE=$(python3 -c "import sys; print(sys.executable)")           \
-      -Donnxruntime_BUILD_UNIT_TESTS=OFF                                              \
-      -Donnxruntime_PREFER_SYSTEM_LIB=ON                                              \
-      -Donnxruntime_BUILD_SHARED_LIB=ON                                               \
-      -DProtobuf_USE_STATIC_LIBS=ON                                                   \
-      ${PROTOBUF_ROOT:+-DProtobuf_LIBRARY=$PROTOBUF_ROOT/lib/libprotobuf.a}           \
-      ${PROTOBUF_ROOT:+-DProtobuf_LITE_LIBRARY=$PROTOBUF_ROOT/lib/libprotobuf-lite.a} \
-      ${PROTOBUF_ROOT:+-DProtobuf_PROTOC_LIBRARY=$PROTOBUF_ROOT/lib/libprotoc.a}      \
-      ${PROTOBUF_ROOT:+-DProtobuf_INCLUDE_DIR=$PROTOBUF_ROOT/include}                 \
-      ${PROTOBUF_ROOT:+-DProtobuf_PROTOC_EXECUTABLE=$PROTOBUF_ROOT/bin/protoc}        \
-      ${RE2_ROOT:+-DRE2_INCLUDE_DIR=${RE2_ROOT}/include}                              \
-      ${BOOST_ROOT:+-DBOOST_INCLUDE_DIR=${BOOST_ROOT}/include}                        \
-      -DCMAKE_CXX_FLAGS="$CXXFLAGS -Wno-unknown-warning -Wno-unknown-warning-option -Wno-error=unused-but-set-variable -Wno-error=deprecated" \
-      -DCMAKE_C_FLAGS="$CFLAGS -Wno-unknown-warning -Wno-unknown-warning-option -Wno-error=unused-but-set-variable -Wno-error=deprecated"
+# Check ROCm build conditions
+if { [ "$ALIBUILD_O2_FORCE_GPU" -ne 0 ] || [ "$ALIBUILD_ENABLE_HIP" -ne 0 ] || command -v /opt/rocm/bin/rocminfo >/dev/null 2>&1; } && \
+   { [ -z "$DISABLE_GPU" ] || [ "$DISABLE_GPU" -eq 0 ]; }; then
+    export ORT_ROCM_BUILD=1
+    : ${ALIBUILD_O2_OVERRIDE_HIP_ARCHS:="gfx906,gfx908"}
+else
+    export ORT_ROCM_BUILD=0
+fi
+# Check CUDA build conditions
+if { [ "$ALIBUILD_O2_FORCE_GPU" -ne 0 ] || [ "$ALIBUILD_ENABLE_CUDA" -ne 0 ] || command -v nvcc >/dev/null 2>&1; } && \
+   { [ -z "$DISABLE_GPU" ] || [ "$DISABLE_GPU" -eq 0 ]; }; then
+    export ORT_CUDA_BUILD=1
```

Comment:
I would also set a default for ALIBUILD_O2_OVERRIDE_CUDA_ARCHS to `sm_86` or `sm_89` architecture for now